core/modules/admin_bar/admin_bar.theme.inc

```diff
@@ -103,21 +99,18 @@ function theme_admin_bar_links($variables) {
         $elements[$path]['#options']['attributes']['class'][] = 'admin-bar-destination';
       }
 
-      $link = l($elements[$path]['#title'], $elements[$path]['#href'], $elements[$path]['#options']);
+      // Append the icon to the sanitized title.
+      $options = $elements[$path]['#options'];
+      $options['html'] = TRUE;
+      $link = l($icon . $title, $elements[$path]['#href'], $options);
     }
     // Handle plain text items, but do not interfere with menu additions.
     elseif (!isset($elements[$path]['#type']) && isset($elements[$path]['#title'])) {
-      if (!empty($elements[$path]['#options']['html'])) {
-        $title = $elements[$path]['#title'];
-      }
-      else {
-        $title = check_plain($elements[$path]['#title']);
-      }
       $attributes = '';
       if (isset($elements[$path]['#options']['attributes'])) {
         $attributes = backdrop_attributes($elements[$path]['#options']['attributes']);
       }
-      $link = '<span' . $attributes . '>' . $title . '</span>';
+      $link = '<span' . $attributes . '>' . $icon . $title . '</span>';
```

Comment:
Quick idea. The title (label text) isn't wrapped in a span on its own, so it would be hard to style it individually. Like hiding the text (via CSS clip) on narrow screens, visually only showing the icon.

Reply:
Thanks, fixed in latest version.